### PR TITLE
[KOGITO-3989] Kogito downstream build on PR

### PR DESCRIPTION
**Pull request by danielezonca**

### CODE REVIEW

File: .ci/kogito-pull-request-config.yaml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  - project: kiegroup/drools
+    build-command:
+      upstream: mvn -e clean install -DskipTests -Dgwt.compiler.skip=true -Dgwt.skipCompilation=true -Denforcer.skip=true -Dcheckstyle.skip=true -Dspotbugs.skip=true -Drevapi.skip=true -Dmaven.wagon.httpconnectionManager.ttlSeconds=25 -Dmaven.wagon.http.retryHandler.count=3
+
+  - project: kiegroup/kogito-runtimes
+    build-command:
+      upstream: |
+        mvn clean install -Dmaven.wagon.httpconnectionManager.ttlSeconds=25 -Dmaven.wagon.http.retryHandler.count=3
+      current: |
+        mvn clean install -Prun-code-coverage -Dmaven.wagon.httpconnectionManager.ttlSeconds=25 -Dmaven.wagon.http.retryHandler.count=3 -Dversion.org.kie7=7.47.0-SNAPSHOT
```
Comment (danielezonca) [inline .ci/kogito-pull-request-config.yaml]: @Ginxo 
I need to make `-Dversion.org.kie7=7.47.0-SNAPSHOT` dynamic. Is it something supported by `github-build-chain`?
The value should be the current snapshot version of Drools (`project.version`).
I can get the value with 
`mvn org.apache.maven.plugins:maven-help-plugin:evaluate -Dexpression=project.version | grep ^7`
or similar but then I probably need to create an env variable and use it.
Can you help me?

Comment (Ginxo) [reply .ci/kogito-pull-request-config.yaml]: we already clarified it

**Comment by danielezonca** [comment]
@Ginxo 
I think the solution proposed to extract DROOLS_VERSION it is not working: the runner seems to append additional parameter at the end of each command producing an invalid result ( [link](https://pipelines.actions.githubusercontent.com/QCyjqVygzNydUxPrYrmU0uleugihcFNAR2pGViDo9ZwPLB7tQ5/_apis/pipelines/1/runs/546/signedlogcontent/3?urlExpires=2020-11-19T16%3A59%3A36.5181491Z&urlSigningMethod=HMACV1&urlSignature=KWLb0jDU0NRir5LEejWL8RDhY2GbeSB6oxxpfEKthjc%3D) ):
```
2020-11-19T15:53:13.1580415Z [INFO]  Execute command 'export DROOLS_VERSION=`mvn org.apache.maven.plugins:maven-help-plugin:evaluate -Dexpression=project.version | grep ^7` -Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn -B' in dir '/home/runner/work/drools/drools/kiegroup_drools'
```
The additional `-Dorg.slf4j.simpleLogger.log.org.apache.maven.cli.transfer.Slf4jMavenTransferListener=warn -B` after the `grep` is the problem.
Any idea?

**Comment by danielezonca** [comment]
I hardcoded the version in the meanwhile to see if the rest of the pipeline is correct

**Review by Ginxo — APPROVED** [review]
amazing!!!

**Comment by Ginxo** [comment]
@rsynek it's also interested on it. @danielezonca do you think we can merge it?

**Comment by danielezonca** [comment]
@Ginxo 
I still need to see a green build and then I'm fine to merge

@rsynek 
Do you want to add OptaPlanner to this job? I can include it directly in this PR using this command https://github.com/kiegroup/kogito-runtimes/pull/873

**Comment by rsynek** [comment]
@danielezonca that would be definitely welcome, but I don't want to hold your PR off; if you want to merge this asap, I can open a separate PR, your call.

Motivation for adding the OptaPlanner here:
Although the check added in https://github.com/kiegroup/kogito-runtimes/pull/873 is sufficient from the OptaPlanner point of view, Drools might want to catch any incompatibilities before merging them. Testing it in kogito-runtimes is a bit late.

**Comment by danielezonca** [comment]
Ok build failed on `kogito-apps` with an error unrelated to drools ([link](https://gist.github.com/danielezonca/16e8166637a2f23b07231d264001090e)), so lets proceed

**Comment by danielezonca** [comment]
@rsynek feel free to change the flow to include OptaPlanner :+1:

**Pull request merged**